src/journal/views.py

```diff
@@ -1929,37 +1930,56 @@ def author_list(request):
     return render(request, template, context)
 
 
-def sitemap(request, issue_id=None):
+def sitemap(request, issue_or_subject_id=None):
     """
     Renders an XML sitemap based on articles and pages available to the journal.
     :param request: HttpRequest object
     :return: HttpResponse object
     """
     try:
         path_parts = None
```

Comment:
There is a fair amount on implicitness here. I understand we guard the logic below using `if request.journal` / ` if request.repository`, but how about we change the URL pattern to take in a site type as well:

`re_path(r'sitemaps/(?P<sitemap_type>issue|repository|journal|book)/(?P<sitemap_id>\d+)/sitemap.xml$', journal_views.sitemap, name='website_sitemap_site_id'),`

or even split the urls into one for each hierarchy types and then pass the arguments to this function with a signature like:
`re_path(r'sitemaps/issue/(?P<issue_id>\d+)/sitemap.xml$', journal_views.sitemap, name='website_sitemap_issue'),`
```
def sitemap(request, issue_id=None, repository_id=None, book_id=None, **other_ids):
    # foobar
```

This second approach makes sense if we also want to split the view logic for each sitemap type or to avoid the multiple logical branches on the view we currently have.